src/Kafka/Consumer.purs

```diff
@@ -346,12 +423,25 @@ type Offsets =
   }
 
 -- | https://github.com/tulios/kafkajs/blob/d8fd93e7ce8e4675e3bb9b13d7a1e55a1e0f6bbf/types/index.d.ts#L660
--- | https://github.com/tulios/kafkajs/blob/d8fd93e7ce8e4675e3bb9b13d7a1e55a1e0f6bbf/types/index.d.ts#L843
--- |
--- | NOTE `Offsets` and `OffsetsByTopicPartition` are exactly the same
 -- |
 -- | `topics: TopicOffsets[]`
+-- |
+-- | NOTE `topics` is optional implementation-wise but marked as required in TS definition
+-- | see comments above `EachBatchPayloadImpl` on `commitOffsetsIfNecessary`
 type OffsetsImpl =
+  Kafka.FFI.Object
+    ()
+    ( topics :: Array TopicOffsetsImpl
+    )
+
+type OffsetsByTopicParition =
+  { topics :: Array TopicOffsets
+  }
```

Comment:
🚫  We should change `topics` to `topicOffsets` to avoid confusion.

Reply:
Same reason as mentioned above to not deviate from `kafkajs`. We could rename things in our project when using the API.